oximeter/db/src/model.rs

```diff
@@ -116,10 +128,33 @@ impl From<Vec<FieldSchema>> for DbFieldList {
     }
 }
 
+const fn default_version() -> u8 {
+    1
+}
+
+// TODO-remove: Remove when DB supports versioned names.
+fn serialize_unversioned_timeseries_name<S>(
+    name: &TimeseriesName,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    serializer.serialize_str(&unversioned_timeseries_name(name))
+}
+
 // The `DbTimeseriesSchema` type models the `oximeter.timeseries_schema` table.
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub(crate) struct DbTimeseriesSchema {
     pub timeseries_name: String,
+    // TODO-remove: Remove the default values and skipped serialization when the
+    // DB supports versioned names.
+    #[serde(default = "default_version", skip_serializing)]
+    pub target_version: u8,
+    // TODO-remove: Remove the default values and skipped serialization when the
+    // DB supports versioned names.
+    #[serde(default = "default_version", skip_serializing)]
+    pub metric_version: u8,
```

Comment:
"256 versions should be enough for anybody!"

Kidding aside, I have no real reason to restrict it other than space, which is a weak justification given that it's only a few bytes of an otherwise large object. ClickHouse will also compress the shit out of these values. I'm fine making it a `u16`.